src/switch_core_media.c

```diff
@@ -15241,7 +15254,14 @@ static void switch_core_media_set_r_sdp_codec_string(switch_core_session_t *sess
 						}
 
 						if (match) {
-							add_audio_codec(map, imp, ptime, buf, sizeof(buf));
+							if (imp->matches_fmtp && map->rm_fmtp) {
+								switch_status_t match_result = imp->matches_fmtp(imp->fmtp, map->rm_fmtp);
+								if (match_result == SWITCH_STATUS_SUCCESS) {
+									add_audio_codec(map, imp, ptime, imp->fmtp, buf, sizeof(buf));
+								}
```

Comment:
Add log here if it didn't match. So we know if something might have broken with this changes

Reply:
probably reset match variable if we failed to add the audio codec